### PR TITLE
DQN extensions

### PR DESCRIPTION
**Pull request by mlaux1**
Solves #24 
Solves #45

### CODE REVIEW

File: rl_blox/algorithm/ddqn.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+        The computed loss for the given minibatch.
+    """
+    obs, reward, action, terminated, next_obs = _extract(batch)
+
+    next_q = q_net(next_obs)
+    indices = jnp.argmax(next_q, axis=1).reshape(-1, 1)
+    next_q_t = q_target(next_obs)
+    next_vals = jnp.take_along_axis(next_q_t, indices, axis=1).squeeze()
+
+    target = jnp.array(reward) + (1 - terminated) * gamma * next_vals
```
Comment (AlexanderFabisch) [inline rl_blox/algorithm/ddqn.py:49]: The loss functions still have the same bug as the original DQN implementation. I'd suggest to merge #113 first.

File: rl_blox/algorithm/nature_dqn.py [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+    optimizer: nnx.Optimizer,
+    batch_size: int = 64,
+    total_timesteps: int = 1e4,
+    gamma: float = 0.99,
+    update_frequency: int = 4,
+    target_update_frequency: int = 1000,
+    q_target_net: MLP | None = None,
+    seed: int = 1,
+    logger: LoggerBase | None = None,
+) -> tuple[MLP, nnx.Optimizer]:
```
Comment (AlexanderFabisch) [inline rl_blox/algorithm/nature_dqn.py:103]: This type hint should be updated.

Comment (mlaux1) [reply rl_blox/algorithm/nature_dqn.py]: Done.

File: rl_blox/algorithm/nature_dqn.py [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+
+            if step % target_update_frequency == 0:
+                q_net = q_target_net
+                q_target_net = nnx.clone(q_net)
+
+        # housekeeping
+        if terminated or truncated:
+            if logger is not None:
+                logger.record_stat(
+                    "return", accumulated_reward, step=step + 1, episode=episode
```
[the remark this answers is not in the document]
Comment (mlaux1) [reply rl_blox/algorithm/nature_dqn.py:220]: Done.

File: rl_blox/algorithm/ddqn.py [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+    gamma: float = 0.99,
+    update_frequency: int = 4,
+    target_update_frequency: int = 1000,
+    q_target_net: MLP | None = None,
+    seed: int = 1,
+    logger: LoggerBase | None = None,
+) -> tuple[MLP, MLP, nnx.Optimizer]:
+    """Deep Q Learning with Experience Replay
+
+    Implements double DQN as originally described by van Hasselt et al. in 2016.
```
[the remark this answers is not in the document]
Comment (mlaux1) [reply rl_blox/algorithm/ddqn.py]: Done.

File: rl_blox/algorithm/ddqn.py [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+                if logger is not None:
+                    logger.record_stat(
+                        "q_loss", q_loss, step=step + 1, episode=episode
+                    )
+                    logger.record_epoch(
+                        "q", q_net, step=step + 1, episode=episode
+                    )
+
+            if step % target_update_frequency == 0:
+                q_target_net = nnx.clone(q_net)
```
[the remark this answers is not in the document]
Comment (AlexanderFabisch) [reply rl_blox/algorithm/ddqn.py]: I think part of the hard update is

```python
if step % target_update_frequency == 0:
```
